io-sim/src/Control/Monad/IOSim/Internal.hs

```diff
@@ -527,6 +533,18 @@ schedule thread@Thread{
     -- ExploreRaces is ignored by this simulator
     ExploreRaces k -> schedule thread{ threadControl = ThreadControl k ctl } simstate
 
+    Fix f k -> do
+      m <- newSTRef (throw NonTermination)
```

Comment:
It's a pitty that we have to throw like this, but I've thought about it and I agree there's no way to throw within the simulator.